llvm/lib/CodeGen/SelectionDAG/LegalizeVectorTypes.cpp

```diff
@@ -4501,7 +4511,19 @@ SDValue DAGTypeLegalizer::SplitVecOp_VECTOR_HISTOGRAM(SDNode *N) {
 }
 
 SDValue DAGTypeLegalizer::SplitVecOp_PARTIAL_REDUCE_MLA(SDNode *N) {
-  return TLI.expandPartialReduceMLA(N, DAG);
+
+  SDLoc DL(N);
+  SDValue Acc = N->getOperand(0);
+  SDValue Input1 = N->getOperand(1);
+
```

Comment:
The code here is assuming that the result/accumulator type has already been split when we get here, which I think is valid. Just to be sure, could you add an assert that `getTypeAction(Acc.getValueType()) != TargetLowering::TypeSplitVector` ?